internal/cmd/beta/server/command/list/list.go

```diff
@@ -0,0 +1,160 @@
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("list server commands: %w", err)
+			}
+			commands := *resp.Items
+			if len(commands) == 0 {
+				p.Info("No commands found for server %s\n", model.ServerId)
+				return nil
+			}
+
+			// Truncate output
+			if model.Limit != nil && len(commands) > int(*model.Limit) {
+				commands = commands[:*model.Limit]
+			}
+			return outputResult(p, model.OutputFormat, commands)
+		},
+	}
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Int64(limitFlag, 0, "Maximum number of entries to list")
+	cmd.Flags().VarP(flags.UUIDFlag(), serverIdFlag, "s", "Server ID")
+
+	err := flags.MarkFlagsRequired(cmd, serverIdFlag)
+	cobra.CheckErr(err)
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command) (*inputModel, error) {
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	limit := flags.FlagToInt64Pointer(p, cmd, limitFlag)
+	if limit != nil && *limit < 1 {
+		return nil, &errors.FlagValidationError{
+			Flag:    limitFlag,
+			Details: "must be greater than 0",
+		}
+	}
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		ServerId:        flags.FlagToStringValue(p, cmd, serverIdFlag),
+		Limit:           limit,
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *runcommand.APIClient) runcommand.ApiListCommandsRequest {
+	req := apiClient.ListCommands(ctx, model.ProjectId, model.ServerId)
+	return req
+}
+
+func outputResult(p *print.Printer, outputFormat string, commands []runcommand.Commands) error {
+	switch outputFormat {
+	case print.JSONOutputFormat:
+		details, err := json.MarshalIndent(commands, "", "  ")
+		if err != nil {
+			return fmt.Errorf("marshal server command list: %w", err)
+		}
+		p.Outputln(string(details))
+
+		return nil
+	case print.YAMLOutputFormat:
+		details, err := yaml.MarshalWithOptions(commands, yaml.IndentSequence(true))
+		if err != nil {
+			return fmt.Errorf("marshal server command list: %w", err)
+		}
+		p.Outputln(string(details))
+
+		return nil
+	default:
+		table := tables.NewTable()
+		table.SetHeader("ID", "STATUS", "STARTED_AT", "FINISHED_AT", "TEMPLATE NAME", "TEMPLATE TITLE")
```

Comment:
Maybe you can change the order of the columns, generally we add name and title after id directly. 

Same comment also for the describe command output :)